src/rgw/setup/rgw.py

```diff
@@ -121,9 +121,19 @@ def config(conf: MappedConf):
         # config phase since data pod starts before server pod.
         # Try HAX endpoint from data pod of same node first & if it doesnt work,
         # from other data pods in cluster
+
+        log_path = Rgw._get_log_dir_path(conf)
+        # Create rgw crash file directory
+        rgw_core_dir = os.path.join(log_path, 'rgw_debug')
+        os.makedirs(rgw_core_dir, exist_ok=True)
+
+        cwd = os.getcwd()
+        os.chdir(rgw_core_dir)
         Rgw._update_hax_endpoint_and_create_admin(conf)
         Log.info('Config phase completed.')
 
+        os.chdir(cwd)
```

Comment:
do we really need to change back to original CWD. Once changed, it will be useful for all rgwadmin cli command usage. If this approach taken, this code becomes one time use, so keeping right at the start would suffice

Reply:
agree. I have removed this code. so all the config phase work will be called from this core directory.